source/GameInterface/Utils/AutoSync/Template/IAutoSyncData.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Common.Messaging;
+using ProtoBuf;
+
+namespace GameInterface.Utils.AutoSync.Template;
+
+/// <summary>
+/// Interface for auto sync data
+/// </summary>
+/// <typeparam name="DataType">Type of data to sync, this will be either a field or property type. This type must also be serializable by protobuf</typeparam>
+public interface IAutoSyncData<DataType>
+{
+    string StringId { get; }
+    DataType Value { get; }
```

Comment:
How is this data's type validated over the network?

Reply:
https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/autosync/source/GameInterface/Utils/AutoSync/Dynamic/DataClassGenerator.cs#L39

Reply:
The DataClassGenerator implements the fields of that interface